File: code/read/hibernate/src/main/java/be/rubus/microstream/performance/hibernate/tests/AllCustomersPaged.java
package be.rubus.microstream.performance.hibernate.tests;

import be.rubus.microstream.performance.QueryInformation;
import be.rubus.microstream.performance.StopWatch;
import be.rubus.microstream.performance.hibernate.Service;
import be.rubus.microstream.performance.hibernate.domain.CustomerEntity;
import be.rubus.microstream.performance.hibernate.util.HibernateUtil;
import be.rubus.microstream.performance.utils.DurationUtil;
import org.hibernate.Session;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.time.Duration;
import java.util.List;

public class AllCustomersPaged {
    public static void main(String[] args) {
        Logger logger = LoggerFactory.getLogger(AllCustomersPaged.class);
        logger.info("Performance run Hibernate");

        StopWatch stopWatch = StopWatch.StartNanoTime();

        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
            DurationUtil.printDuration(logger, "Database Access", stopWatch.stop());

            Service service = new Service(session);

            logger.info("Execute Query - all customers Paged");
            QueryInformation<List<CustomerEntity>> listQueryInformation = service.allCustomersPaged();

            DurationUtil.printDuration(logger, "Query Execution", listQueryInformation.getElapsedNanoSeconds());

        }

    }
}

File: code/read/exposed/src/main/kotlin/be/rubus/microstream/performance/exposed/tests/AllCustomersPaged.kt
package be.rubus.microstream.performance.exposed.tests

import be.rubus.microstream.performance.StopWatch
import be.rubus.microstream.performance.exposed.DBSettings
import be.rubus.microstream.performance.exposed.Service
import be.rubus.microstream.performance.exposed.readonlyTransaction
import be.rubus.microstream.performance.utils.DurationUtil
import org.slf4j.LoggerFactory

fun main() {
    val logger = LoggerFactory.getLogger("AllCustomersPaged")
    logger.info("Performance run Exposed")

    val stopWatch: StopWatch = StopWatch.start()
    DBSettings.db
    DurationUtil.printDuration(logger, "Database Access", stopWatch.stop())

    logger.info("Execute Query - all customers Paged")
    val listQueryInformation = readonlyTransaction(DBSettings.db) {
        Service.allCustomersPaged()
    }

    DurationUtil.printDuration(logger, "Query Execution", listQueryInformation.elapsedNanoSeconds)


}

